### ISSUE TITLE
Export location of nearby missing pictures to GPX file

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#953]
The "Nearby missing pictures" map exists, but many people would probably prefer to see the location of their favorite maps app such as OsmAnd/Maps.me/etc.

So, it would be great to have an "Export to GPX" button that asks you for the desired radius, and generates a GPX file containing all of the latitudes/longitudes. KML/KMZ formats could be a nice addition.

The KMZ file for the whole world (generated by my script https://github.com/nicolas-raoul/wikidata-missing-pictures-offline) takes 100MB, so a default radius of 100km sounds reasonable.

Another advantage is that the file can then be used offline, for instance if you only have WiFi at the hotel r in your home country. Being able to select the center would be a nice addition, for instance letting you download all of Cuba's points while preparing your trip at home in Pakistan.

I tag "beginner-friendly" because while it takes some time implementing, I believe there is nothing really difficult technically. The Wikidata query is already present in the code, and GPX format can be copied from the Github project above.

KMZ file seen on Maps.me:
![screenshot_20171113-104437](https://user-images.githubusercontent.com/99590/32706337-38ca7296-c860-11e7-8d1a-e4cda8c7e23e.png)

**Comment by nsaiisasidhar** [comment]
Hi, can I work on this enhancement.

**Comment by maskaravivek** [comment]
@nsaiisasidhar Awesome thanks! Go ahead with it. :)

**Comment by maskaravivek** [comment]
@nsaiisasidhar Awesome thanks! Go ahead with it. :)

**Comment by nicolas-raoul** [comment]
Hi @nsaiisasidhar any progress with this? :-)

You will need to write some Java code that does the same as my shell script, called by a button somewhere (for instance an action bar button in "Nearby Places" followed by a confirmation popup saying "You are about to download a large GPX file of nearby places. This can take a few minutes and a lot of network data. Continue/Cancel").

**Comment by nsaiisasidhar** [comment]
@nicolas-raoul Hi, I'm working on it. I'll make the PR in few days. Thank you for the information.

**Comment by mashawan** [comment]
Hello, I notice this is unassigned but may be being worked on.  I have worked in Geo systems in the past and this is of interest to me.  Could I take this or is this still in active development?

**Comment by mashawan** [comment]
<bump> ;-)

**Comment by mashawan** [comment]
--bump-- ;-)

**Comment by nicolas-raoul** [comment]
@nsaiisasidhar Yes, if you get the results as CSV then feel free to use regular expressions to make that into GPX. Purists might argue that you should get JSON and parse it, but that would take more memory, so CSV is OK I guess.
Indeed, please use a radius query if you can, thanks! :-)
Sorry for my late reply. I guess this is your current code https://github.com/nsaiisasidhar/apps-android-commons/commits/953-Export-location-of-nearby-missing-pictures-to-GPX-file ?

In what state is your code? Is it ready, or in progress?
If in progress: Do you want to continue working on it, or do you can @mashawan continue your work?

Thanks!

**Comment by Rishavgupta12345** [comment]
I don't think this issue is relevant now. Please review and close this issue

**Comment by nicolas-raoul** [comment]
@Rishavgupta12345 This issue is very relevant, we still need this feature.

**Comment by misaochan** [comment]
Hi @Rishavgupta12345 , could you elaborate on why you feel this issue is irrelevant now?

**Comment by kanahia1** [comment]
Hey @nicolas-raoul, Can I work on this issue ?

**Comment by kanahia1** [comment]
Shared a pull request here https://github.com/commons-app/apps-android-commons/pull/5645 

I am open to suggestions 🙌
